starlark/stable/pipeline.star

```diff
@@ -22,7 +22,7 @@ def push(**kwargs):
     Example usage: `action(tasks=["test"], on=push(branches=["master"]))`
     """
 
-    return p.Condition(push=p.PushCondition(**kwargs))
+    return p.Condition(git=p.GitCondition(push=p.GitPushCondition(**kwargs)))
```

Comment:
I don't think this form works right now. I get this:

```
time="2020-12-02T18:02:01Z" level=error msg="Failed to match action #7: no condition is set
```

That's returned by the Match function: https://github.com/mesosphere/dispatch/blob/dev/pkg/pipeline/v0/condition.go#L55

But we can use the backwards compatible ones and it should work.